src/dist/replication/lib/bulk_load/replica_bulk_loader.cpp

```diff
@@ -309,28 +309,102 @@ error_code replica_bulk_loader::download_sst_files(const std::string &app_name,
 }
 
 // ThreadPool: THREAD_POOL_REPLICATION
-error_code replica_bulk_loader::parse_bulk_load_metadata(const std::string &fname,
-                                                         /*out*/ bulk_load_metadata &meta)
+error_code replica_bulk_loader::parse_bulk_load_metadata(const std::string &fname)
 {
-    // TODO(heyuchen): TBD
-    // read file and parse file content as bulk_load_metadata
+    if (!utils::filesystem::file_exists(fname)) {
+        derror_replica("file({}) doesn't exist", fname);
+        return ERR_FILE_OPERATION_FAILED;
+    }
+
+    int64_t file_sz = 0;
+    if (!utils::filesystem::file_size(fname, file_sz)) {
+        derror_replica("get file({}) size failed", fname);
+        return ERR_FILE_OPERATION_FAILED;
+    }
+
+    std::shared_ptr<char> buf = utils::make_shared_array<char>(file_sz + 1);
```

Comment:
Use std::string. Using `std::shared_ptr<char>` to represent a buffer is a bad practice.
If you want it to convert to a dsn::blob, try out this way:

```cpp
std::string buf('\0', file_sz+1);
fin.read(&buf[0], file_sz);
blob bb = blob::create_from_bytes(std::move(buf));
```

This involves only one memory creation, no copy.